src/components/TableToolbar.js

```diff
@@ -105,8 +105,9 @@ class TableToolbar extends React.Component {
             data: row.data.map(column => {
               i += 1;
 
-              // if we have a custom render, we must grab the actual value from data
-              return typeof column === 'object' ? data[row.dataIndex].data[i] : column;
+              // if we have a custom render, which will appear as a react element, we must grab the actual value from data
+              // TODO: Create a utility function for checking whether or not something is a react object
+              return typeof column === 'object' && column !== null && !Array.isArray(column) ? data[row.dataIndex].data[i] : column;
```

Comment:
So, this doesn't resolve the original bug.
Here the "faulty" column is
```javascript
{$$typeof: Symbol(react.element), key: null, ref: null, props: {…}, type: ƒ, …}
```
which is not of type array, so we do the wrong calculation again.

Reply:
Not sure which bug you mean. The only bug I've been able to reproduce is one in which, with `useDisplayedRowsOnly` is set to `true`, some of the data is in array form, the table is sorted, and then the csv is downloaded. In that case, the arrays were not returning the correct values and it showed up to me as exactly the mismatched downloaded csv in your example. It definitely fixes this exact set of circumstances. If there is another set of circumstances, then I would need those steps spelled out and demonstrated in another issue.

Reply:
Sorry, my bad. I just checked out your branch and everything indeed works!

Just wanted to say thanks for this library, it saves a lot of boilerplate and was really easy to set up.

Reply:
Thank you, on behalf of the community! Glad to hear it's working for you.